Assets/Scripts/NPC/NpcMovement.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections;
+using FrameWork.Extensions;
+using FrameWork.GridSystem;
+using UnityEngine;
+
+namespace NPC
+{
+    public class NpcMovement : MonoBehaviour
+    {
+        [SerializeField, Range(1, 10)] private float movementSpeed;
+        
+        private int _enumLength;
+        private int _gridIndex;
+
+        private bool _moving;
+        private void Awake() => _enumLength = Enum.GetNames(typeof(NpcGridpoints)).Length;
```

Comment:
```suggestion
        private bool _moving;
        
        private void Awake() => _enumLength = Enum.GetNames(typeof(NpcGridpoints)).Length;
```
Mijn brein vind dit niet leuk.